lib/src/render/documentation_renderer.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:dartdoc/src/tuple.dart';
+import 'package:html/parser.dart' show parse;
+import 'package:markdown/markdown.dart' as md;
+
+abstract class DocumentationRenderer {
+  Tuple2<String, String> render(List<md.Node> nodes, bool processFullDocs);
+}
+
+class DocumentationRendererHtml extends DocumentationRenderer {
+  @override
+  Tuple2<String, String> render(List<md.Node> nodes, bool processFullDocs) {
```

Comment:
The old code had an optimization that avoided calculating links for parts of the documentation that would never be rendered (if it is only used in a oneLineDoc).  This used to make a very noticeable difference on dartdoc's performance.  It looks as though that optimization has been removed.

If it doesn't matter anymore we can probably drop the flag.

Reply:
If I understand correctly, I think we are still honoring that flag in `MarkdownDocument.parseMarkdownText`, it stops collecting markdown nodes if the flag is false. It's just a little harder to follow now because half the code path is there and the other half moved to the renderer. `Documentation._renderDocumentation` calls both methods in succession, passing the same value for the flag, so it should still be doing this optimization.